pkg/reconciler/dns/health.go

```diff
@@ -0,0 +1,66 @@
+package dns
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"io"
+
+	v1 "github.com/kuadrant/kcp-glbc/pkg/apis/kuadrant/v1"
+	"github.com/kuadrant/kcp-glbc/pkg/dns"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	healthEndpointAnnnotation = "kuadrant.dev/health-endpoint"
+)
+
+func (c *Controller) ReconcileHealthChecks(ctx context.Context, ingress *networkingv1.Ingress, dnsRecord *v1.DNSRecord) error {
+	if ingress.Annotations == nil {
+		return nil
+	}
+
+	healthEndpoint := ""
+	ok := false
+	if healthEndpoint, ok = ingress.Annotations[healthEndpointAnnnotation]; !ok {
+		return nil
+	}
```

Comment:
re. https://github.com/Kuadrant/kcp-glbc/pull/52/files#r837228106

```suggestion
	}	if healthEndpoint, ok = dnsRecord.Annotations[HealthEndpointAnnnotation]; !ok {
		return nil
	}
```